kuma/javascript/src/utils.js

```diff
@@ -20,3 +20,60 @@ export function getCookie(name) {
     }
     return cookieValue;
 }
+
+/**
+ * Creates a new XMLHttpRequest, sets the specified request
+ * headers if specified, and returns the new XMLHttpRequest object
+ * @param {String} method - The XMLHttpRequest method to use
+ * @param {String} url - The URL endpoint
+ * @param {Object} requestHeaders - Request headers as an Object
+ *
+ * Example requestHeaders
+ * -----------------------
+ * {
+ *   'X-Requested-With': 'XMLHttpRequest',
+ *   'Content-type': 'application/x-www-form-urlencoded'
+ * }
+ *
+ * @returns new XMLHttpRequest object
+ */
+export function initAjaxRequest(method, url, requestHeaders) {
```

Comment:
looks like we're using `fetch` in other parts of the codebase, which imo has a saner API, so I'd vote going with that instead. If older browser support is the issue, I'd vote for loading a polyfill on those (relatively few?) devices.